### PR TITLE
chore: add .github/renovate.json5

### PR DESCRIPTION
**Pull request by harish-sethuraman**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

fixes https://github.com/eslint/rewrite/issues/30

#### What changes did you make? (Give an overview)

add renovate config from eslint/eslint

#### Related Issues

https://github.com/eslint/rewrite/issues/30

#### Is there anything you'd like reviewers to focus on?

- confirm if we have enabled renovate for this `rewrite` repository
- documentation says renovate works out of the box in monorepos so no extra configuration is needed?

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+			description: "Group wdio packages into a single PR.",
+			groupName: "wdio",
+			matchPackagePrefixes: ["@wdio"],
+		},
+		{
+			description: "Group metascraper packages into a single PR.",
+			groupName: "metascraper",
+			matchPackagePrefixes: ["metascraper"],
+		},
+	],
```
Comment (nzakas) [inline .github/renovate.json5:41]: I don't think we need this as we don't use metascraper here.

File: .github/renovate.json5 [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		{
+			description: "Use the deps:npm label for npm manager packages (this means Renovate's npm manager).",
+			addLabels: ["deps:npm"],
+			matchManagers: ["npm"],
+		},
+		{
+			description: "Group Babel packages into a single PR.",
+			groupName: "babel",
+			matchPackagePrefixes: ["@babel", "babel-"],
+		},
```
Comment (nzakas) [inline .github/renovate.json5:30]: This can be removed as we aren't using Babel in this repo

Comment (harish-sethuraman) [reply .github/renovate.json5]: removed `@wdio` as I didn't see it in the repo either

File: .github/renovate.json5 (hunk 1 above)
Comment (harish-sethuraman) [reply .github/renovate.json5]: done

**Review by aladdin-add — APPROVED** [review]
there is an issue to discuss reusing the config: https://github.com/eslint/eslint/issues/17856. But until it's done, we can be copied over for now.

LGTM. Would like @nzakas to verify before merging.

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**